### ISSUE TITLE
Whitelisting required

### ISSUE THREAD

**Issue by danielmartins** [ridrum/cloudwatch-backend-for-statsd#1]
Hi, 

I'm trying to use your package to send metrics to cloudwatch. 

In the docs, doesn't explain the need to declare all metrics(in whitelist parameter) that should be sent. If I dind't describe then none metric is sent at all. 

Can I describe a wildcard to send all received metrics? like * or __all__ ?

I think the **processKeyForNames** is not working too.

I'm using the following configuration:

```
{
    backends: [ "cloudwatch-backend-for-statsd" ],
    cloudwatch:
    {
        accessKeyId: '--',
        secretAccessKey:'--',
        region:"us-east-1",
        processKeyForNames: true,
        whitelist: [
            "App.Controller.Action.Request"
        ]
    },
    debug: true,
    dumpMessages: true
}
```

sending metric manually
```
echo "App.Controller.Action.Request:1|c" | nc -u -w0 127.0.0.1 8125
```

Still sending the default namespace **AwsCloudWatchStatsdBackend**
```
21 Sep 18:08:45 - [1] reading config file: /statsd/config.js
21 Sep 18:08:45 - DEBUG: Loading server: ./servers/udp
21 Sep 18:08:45 - server is up
21 Sep 18:08:45 - DEBUG: Loading backend: cloudwatch-backend-for-statsd
Starting cloudwatch reporter instance in region: us-east-1
21 Sep 18:08:46 - DEBUG: App.Controller.Action.Request:1|c
Flushing metrics at 2018-09-21T18:08:55.000Z
Counter metrics created : [{"MetricName":"App.Controller.Action.Request","Unit":"Count","Timestamp":"2018-09-21T18:08:55.000Z","Value":1}] with namespace : AwsCloudWatchStatsdBackend
Timer metrics created : [] with namespace : AwsCloudWatchStatsdBackend
Gauge metrics created : [] with namespace : AwsCloudWatchStatsdBackend
Set metrics created : [] with namespace : AwsCloudWatchStatsdBackend
Flushing metrics at 2018-09-21T18:09:05.000Z
```

Am I missing something ?

**Comment by jupier** [comment]
Hi daniel,

thank you for your report !

you're right there is something missing. if you dont declare all the white
listed metrics none will be send... sorry for this issue. I will do a quick
fix in the next hours and ping when it s done 😜

Dont hesitate if you find other issues...

**Comment by jupier** [comment]
Hi @danielmartins ! 

I've just published a new version of the package (0.0.7).

The whitelist and processKeyForNamespace should be fixed.

For now, if you don't pass a whitelist attribute, all the metrics will be send to cloudwatch. I also added regexp support for the whitelist values (please check the readme for more information).

I also changed the readme for the processKey stuff cause there were an error. The parameter name is not _processKeyForNames_ but _processKeyForNamespace_. Your statd config file should be : 

```
{
    backends: [ "cloudwatch-backend-for-statsd" ],
    cloudwatch:
    {
        accessKeyId: '--',
        secretAccessKey:'--',
        region:"us-east-1",
        processKeyForNamespace: true,
        whitelist: ....
    },
    debug: true,
    dumpMessages: true
}
```

Don't hesitate if you still have some issues

:neckbeard:
